moto/rds/models.py

```diff
@@ -3947,6 +3981,108 @@ def describe_db_log_files(self, db_instance_identifier: str) -> List[DBLogFile]:
+        filters: Any = None,
+    ) -> List[BlueGreenDeployment]:
+        bg_deployments = self.bluegreen_deployments
+        if blue_green_deployment_identifier:
+            filters = merge_filters(
+                filters,
+                {
+                    "blue-green-deployment-identifier": [
+                        blue_green_deployment_identifier
+                    ]
+                },
+            )
+        if filters:
+            bg_deployments = self._filter_resources(
+                bg_deployments, filters, BlueGreenDeployment
+            )
+        if blue_green_deployment_identifier and not bg_deployments:
+            raise BlueGreenDeploymentNotFoundFault(blue_green_deployment_identifier)
+        return list(bg_deployments.values())
+
+    def switchover_blue_green_deployment(
+        self,
+        blue_green_deployment_identifier: str,
+        switchover_timeout: Optional[int] = 300,
+    ) -> BlueGreenDeployment:
+        if blue_green_deployment_identifier not in self.bluegreen_deployments:
+            raise BlueGreenDeploymentNotFoundFault(blue_green_deployment_identifier)
+
+        bg_deployment = self.bluegreen_deployments[blue_green_deployment_identifier]
+
+        if bg_deployment.status != "AVAILABLE":
+            raise InvalidBlueGreenDeploymentStateFault(blue_green_deployment_identifier)
+
+        bg_deployment_before_switch = copy.copy(bg_deployment)
+
+        bg_deployment.switchover()
+        self.bluegreen_deployments[blue_green_deployment_identifier] = bg_deployment
+
+        bg_deployment_before_switch.set_status(
+            bg_deployment.SupportedStates.SWITCHOVER_IN_PROGRESS
+        )
+
+        return bg_deployment_before_switch
+
+    def delete_blue_green_deployment(
+        self, blue_green_deployment_identifier: str, delete_target: bool = False
+    ) -> BlueGreenDeployment:
+        if blue_green_deployment_identifier not in self.bluegreen_deployments:
+            raise BlueGreenDeploymentNotFoundFault(blue_green_deployment_identifier)
+        bg_deployment = self.bluegreen_deployments.pop(blue_green_deployment_identifier)
+
+        if (
+            delete_target
+            and bg_deployment.status
+            != bg_deployment.SupportedStates.SWITCHOVER_COMPLETED.name
+        ):
+            if self._is_cluster(bg_deployment.target):
+                cluster = self.describe_db_clusters(
+                    db_cluster_identifier=bg_deployment.target
+                )[0]
+                for member in cluster.members:
+                    self.delete_db_instance(
+                        db_instance_identifier=member.db_instance_identifier
+                    )
+                self.delete_db_cluster(cluster_identifier=cluster.db_cluster_identifier)
+            else:
+                instance = self.describe_db_instances(
+                    db_instance_identifier=bg_deployment.target
+                )[0]
```

Comment:
Using `self.find_db_from_id` here would obviate the need for manual array indexing.

(Yes, we should probably standardize our `find_*` helpers, but that is out of scope for this PR.)